modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -46,20 +49,56 @@ export default function DatePicker({
 	const maskRef = useRef();
 	const [expanded, setExpand] = useState(false);
 
-	const {clayFormat, momentFormat, placeholder} = useMemo(() => {
-		const momentFormat = moment()
-			.locale(locale ?? defaultLanguageId)
-			.localeData()
-			.longDateFormat('L');
+	const {
+		clayFormat,
+		isDateTime,
+		momentFormat,
+		placeholder,
+		serverFormat,
+		use12Hours,
+	} = useMemo(() => {
+		let use12Hours = false;
+
+		const isDateTime = type === 'date_time';
+		const momentLocale = moment().locale(locale ?? defaultLanguageId);
+		const dateFormat = momentLocale.localeData().longDateFormat('L');
+		const time = momentLocale.localeData().longDateFormat('LT');
+
+		let momentFormat = dateFormat;
+
+		if (isDateTime) {
+			const [hourFormat] = time.split(NOT_LETTER_REGEX, 1);
 
-		const clayFormat = momentFormat
+			const formattedTime =
+				hourFormat.length === 1
+					? hourFormat[0] === 'H'
+						? `H${time}`
+						: `h${time}`
+					: time;
+
+			momentFormat = `${dateFormat} ${formattedTime}`;
+			use12Hours = time.endsWith('A');
+		}
+
+		const clayFormat = dateFormat
 			.replace('YYYY', 'yyyy')
 			.replace('DD', 'dd');
 
-		const placeholder = clayFormat.replace(/\w/g, '_');
+		const placeholder = momentFormat.replace(/\w/g, '_');
```

Comment:
Hi @rebsilva! I saw that this line was not inserted by you, but could you create a constant with this regex `/\w/g` just to keep the pattern you've been doing? :blush:

Reply:
Sure, I will do that!